### PR TITLE
Test Messenger Adapters

### PR DESCRIPTION
**Pull request by michelbl**
Connected to https://app.asana.com/0/481142309329793/660479534899917

Because we were all ashamed of the poor coverage of the `MessengerAdapter`. The testing is not perfect yet (only the messenger adapter, one very big test) but as Martin Fowler says, an imperfect test suite is better than no test.

This PR introduces the following changes:
* dev dependency on `sinon` is put in the root `package.json`
* `MessengerAdapter.handleRequest` sends a response 200 and returns *after* all the events have been processed. See https://github.com/Botfuel/botfuel-dialog/pull/177/files#diff-e275d0323410e4245b0dde03c9bac553R112 This is better for testing.
* It took me some time to add the flow annotation but it exposed a bug that is fixed here: https://github.com/Botfuel/botfuel-dialog/pull/177/files#diff-e275d0323410e4245b0dde03c9bac553R145

### CODE REVIEW

**Comment by michelbl** [comment]
This PR increases coverage by 2.33%, yet codecov complains... (because the coverage of the *patch* is below the project average)

File: packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js [hunk 1]
```diff
@@ -87,3 +95,126 @@ describe('MessengerAdapter', () => {
...
+
+describe('MessengerAdapter request handling', () => {
+  test('should handle a request from messenger', async () => {
+    // Data definitions
+
+    const profile = {
+      firstName: 'Sample',
+      lastName: 'User',
+    };
+    const userMessageUuid = '254a66a3-9282-4e4f-bd86-e18a376d5bc4';
```
Comment (windkomo) [inline packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js]: where do those uuids come from?
if they are random and do not affect tests, maybe we could use something like `uuidv4()`?

File: packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js [hunk 2]
```diff
@@ -87,3 +95,126 @@ describe('MessengerAdapter', () => {
...
+    const botMessageUuid = 'cbad2bc9-070c-4b37-b4be-190904fc506e';
+    const handlingTimestamp = 1526398451040;
```
Comment (windkomo) [inline packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js]: Maybe we can use Date.now() to generate a timestamp and for `responseTimestamp` add some delay to this timestamp?

File: packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js [hunk 3]
```diff
@@ -87,3 +95,126 @@ describe('MessengerAdapter', () => {
...
+    const responseTimestamp = 1526477455343;
+    const userMessageText = 'hello ca va?';
+    const testRequest = {
+      body: {
+        object: 'page',
+        entry: [
+          {
+            id: '<PAGE_ID>',
+            time: 1458692752478,
```
Comment (windkomo) [inline packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js]: where does this value come from?

File: packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js [hunk 4]
```diff
@@ -87,3 +95,126 @@ describe('MessengerAdapter', () => {
...
+            messaging: [
+              {
+                sender: {
+                  id: TEST_USER,
+                },
+                recipient: {
+                  id: '1803703159928348',
```
Comment (windkomo) [inline packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js]: can we put anything here?

File: packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js [hunk 5]
```diff
@@ -87,3 +95,126 @@ describe('MessengerAdapter', () => {
...
+                },
+                timestamp: 1526396479959,
+                message: {
+                  mid: 'mid.$cAAYyhkMB6E1plZcx11jZFHo5moFm',
+                  seq: 38631,
+                  text: userMessageText,
```
Comment (windkomo) [inline packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js]: where do `mid` and `seq` come from?

File: packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js (hunk 2 above)
Comment (michelbl) [reply packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js]: +1

File: packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js (hunk 3 above)
Comment (michelbl) [reply packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js]: A real message sent on messenger.

File: packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js (hunk 4 above)
Comment (michelbl) [reply packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js]: I replaced it by something more explicit.

File: packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js (hunk 5 above)
Comment (michelbl) [reply packages/botfuel-dialog/tests/adapters/messenger-adapter.test.js]: A real message sent on messenger.

**Pull request merged**